source/Halibut/Transport/PollingClient.cs

```diff
@@ -95,5 +117,56 @@ void ExecutePollingLoop(object ignored)
                 }
             }
         }
+
+        async Task ExecutePollingLoopAsyncCatchingExceptions()
+        {
+            try
+            {
+                await ExecutePollingLoopAsync();
+            }
+            catch (Exception)
+            {
+                // We may get errors about the workingCancellationTokenSource being used when it is disposed, we don't care about that.
+            }
+        }
+        async Task ExecutePollingLoopAsync()
+        {
+            var retry = createRetryPolicy();
+            var sleepFor = TimeSpan.Zero;
+            while (!workingCancellationTokenSource.Token.IsCancellationRequested)
```

Comment:
I think this will cause exceptions we don't want.

If you create the token right at the start (maybe a member variable), then you can query the `IsCancellationRequested` property all you want. Even if the `workingCancellationTokenSource` has been disposed.

But if you query `workingCancellationTokenSource.Token.IsCancellationRequested` after it has been disposed, then it will throw an exception (as it has to make a new one, and the "shop is closed").